crates/astria-composer/src/searcher.rs

```diff
@@ -0,0 +1,353 @@
+        let inner = Provider::connect(url).await?;
+        Ok(Self {
+            inner,
+        })
+    }
+
+    /// Wrapper around [`Provider::get_net_version`] with a 1s timeout.
+    async fn get_net_version(&self) -> eyre::Result<String> {
+        use ethers::providers::Middleware as _;
+        tokio::time::timeout(Duration::from_secs(1), self.inner.get_net_version())
+            .await
+            .wrap_err("request timed out")?
+            .wrap_err("RPC returned with error")
+    }
+}
+
+/// A thin wrapper around [`sequencer_client::Client`] to add timeouts.
+///
+/// Currently only provides a timeout for `abci_info`.
+#[derive(Clone)]
+struct SequencerClient {
+    inner: sequencer_client::HttpClient,
+}
+
+impl SequencerClient {
+    #[instrument]
+    fn new(url: &str) -> eyre::Result<Self> {
+        let inner = sequencer_client::HttpClient::new(url)
+            .wrap_err("failed to construct sequencer client")?;
+        Ok(Self {
+            inner,
+        })
+    }
+
+    /// Wrapper around [`Provider::get_net_version`] with a 1s timeout.
+    async fn abci_info(self) -> eyre::Result<abci::response::Info> {
+        use sequencer_client::Client as _;
+        tokio::time::timeout(Duration::from_secs(1), self.inner.abci_info())
+            .await
+            .wrap_err("request timed out")?
+            .wrap_err("RPC returned with error")
+    }
+}
+
+impl Searcher {
+    /// Constructs a new Searcher service from config.
+    pub async fn from_config(cfg: &Config) -> eyre::Result<Self> {
+        // connect to eth node
+        let eth_client = EthClient::connect(&cfg.execution_url)
+            .await
+            .wrap_err("failed connecting to geth")?;
+
+        // connect to sequencer node
+        let sequencer_client = SequencerClient::new(&cfg.sequencer_url)
+            .wrap_err("failed constructing sequencer client")?;
+
+        let rollup_chain_id = cfg.chain_id.clone();
+        let (status, _) = watch::channel(Status::default());
+
+        Ok(Searcher {
+            eth_client,
+            sequencer_client,
+            rollup_chain_id,
+            status,
+            conversion_tasks: JoinSet::new(),
+            submission_tasks: JoinSet::new(),
+        })
+    }
+
+    pub(crate) fn subscribe_to_state(&self) -> watch::Receiver<Status> {
+        self.status.subscribe()
+    }
+
+    /// Serializes and signs a sequencer tx from a rollup tx.
+    async fn handle_pending_tx(&mut self, rollup_tx: Transaction) -> eyre::Result<()> {
+        let chain_id = self.rollup_chain_id.clone();
+
+        self.conversion_tasks.spawn_blocking(move || {
+            // FIXME: Needs to be altered when nonces are implemented in the sequencer
+            // For now, each transaction is transmitted from a new account with nonce 0
+            let sequencer_key = ed25519_consensus::SigningKey::new(rand::thread_rng());
+            let nonce = Nonce::from(0);
```

Comment:
this will break with #206 as we will require a funded account -  this is fine for now but just wanted to point that out, composer will need to updated to use a funded sequencer account

Reply:
Created #215 and added a link to the GH issue to the `FIXME`.